### PR TITLE
Refinements to the workflow #1236

### LINKED ISSUE #1236: Refinements to the workflow
I propose following changes to the workflow:

**Issue life cycle:**
* An issue in `status.toMerge` state should be verified by PM (indicated by the `status.mergeApproved` label) before it is merged. If the PM asks for further changes, the issue goes back to `status.ongoing`
Reason: we can't rely on just one peer reviewer to ensure the code quality.  
* Senior developers and above should not take lower priority issues when there are open higher priority issues. Committers are encouraged to do the same.
Reason: we have very limited man power. We should spend it on high priority issues. That's the whole purpose of having priorities.

**Team Lead responsibilities:**
* Ensure the Milestone plan should be ready at the beginning of the milestone.
Reason: Developers need more practice on planning and estimation. 

**Coding style:**
* All public members and non-trivial private members should have header comments.
Reason: Header comments help to ensure members are well-defined. e.g. conformance to Single Responsibility Principle.

### PR DESCRIPTION
**Pull request by dariusf**
Fixes #1236

### CODE REVIEW

File: docs/process.md [hunk 1]
```diff
@@ -28,6 +28,24 @@ You may not need these at the beginning, but be prepared to learn them along the
...
+## PR Lifecycle
+
+- A PR corresponding to an issue is created and labelled as `ongoing`.
+- It is labelled as `toReview` when the developer finishes their work.
+- The reviewer labels it as `toMerge` when satisfied, or `ongoing` if further changes have to be made.
+- The PM labels it as `mergeApproved` when satisfied, or `ongoing` if further changes have to be made.
+- The PR is merged.
+
+A PR should be merged within one release cycle, two at the most. The onus is on the developer to remind the reviewer to keep things moving. PRs still open after two weeks will be closed.
+
```
Comment (damithc) [inline docs/process.md]: `will be closed` --> `may be closed without merging`

**Comment by damithc** [comment]
Under instructions for release, we can mention that the team lead should check open PRs and remind dev to finish by next cycle, or close if the PR is stale.

**Comment by dariusf** [comment]
Added.

File: docs/process.md [hunk 2]
```diff
@@ -88,23 +109,25 @@ An exhaustive list of things to look out for really cannot be given here, but th
...
 - Method names and comments not outdated as a result of changes
-- `Optional` values checked before use
 - Static imports used effectively
+- `Optional` values checked before use
 - Thread-safe
 - Meaningful test cases
 - Proper synchronisation in GUI tests (no `Thread.sleep` or `PlatformEx.waitOnFxThread` without good reason)
 
-When you are satisfied with the quality of the changes in the PR, change the label to `toMerge`.
+When you are satisfied with the quality of the changes in the PR, change the label to `toMerge`. At this point it will be verified.
```
Comment (damithc) [inline docs/process.md]: `approved` (to be consistent with the label)?

File: docs/process.md [hunk 3]
```diff
@@ -113,6 +136,7 @@ If the PR was from a contributor, you can merge it at this point.
 
 - Put completed issues under current milestone
 - Remove milestones for issues which weren't finished
+- Determine if open PRs can be merged by the next milestone, and close them otherwise
```
Comment (damithc) [inline docs/process.md]: More specifically, (a) add a comment in open PRs to close in the next milestone (b) close PRs that have not been making progress in spite of reminders.

Comment (dariusf) [inline docs/process.md]: Okay, that's more flexible.

**Pull request merged**